src/modules/flow/http-server/http-server.c

```diff
@@ -0,0 +1,622 @@
+            break;
+        case SOL_HTTP_PARAM_HEADER:
+            if (streq(value->value.key_value.key, HTTP_HEADER_ACCEPT)) {
+                if (strstr(value->value.key_value.value, HTTP_HEADER_CONTENT_TYPE_JSON))
+                    send_json = true;
+            }
+            break;
+        default:
+            break;
+        }
+    }
+
+    if (send_json) {
+        r = snprintf(str, sizeof(str), "{\n\t\"%s\": \"%s\"\n}", mdata->path,
+            mdata->value.s);
+    } else {
+        r = snprintf(str, sizeof(str), "%s", mdata->value.s);
+    }
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
+        HTTP_HEADER_CONTENT_TYPE, (send_json) ? HTTP_HEADER_CONTENT_TYPE_JSON : HTTP_HEADER_CONTENT_TYPE_TEXT));
+    SOL_INT_CHECK_GOTO(r, != true, end);
+
+    r = sol_buffer_set_slice(&response.content, sol_str_slice_from_str(str));
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    r = sol_http_server_send_response(request, &response);
+    SOL_INT_CHECK_GOTO(r, < 0, end);
+
+    if (method == SOL_HTTP_METHOD_POST) {
+        sol_flow_send_string_packet(mdata->node, SOL_FLOW_NODE_TYPE_HTTP_SERVER_STRING__OUT__OUT, mdata->value.s);
+    }
+
+end:
+    sol_buffer_fini(&response.content);
+    sol_http_param_free(&response.param);
+
+    return r;
+}
+
+static int
+string_process(struct sol_flow_node *node, void *data, uint16_t port, uint16_t conn_id, const struct sol_flow_packet *packet)
+{
+    struct http_data *mdata = data;
+    const char *val;
+    int r;
+
+    r = sol_flow_packet_get_string(packet, &val);
+    SOL_INT_CHECK(r, < 0, r);
+
+    r = sol_http_server_set_last_modified(server, mdata->path, time(NULL));
+    SOL_INT_CHECK(r, < 0, r);
+
+    free(mdata->value.s);
+    mdata->value.s = strdup(val);
+    SOL_NULL_CHECK(mdata->value.s, -ENOMEM);
+
+    return 0;
+}
+
+static void
+string_close(struct sol_flow_node *node, void *data)
+{
+    struct http_data *mdata = data;
+
+    free(mdata->value.s);
+    stop_server(mdata);
+}
+
+static int
+string_open(struct sol_flow_node *node, void *data, const struct sol_flow_node_options *options)
+{
+    int r;
+    struct http_data *mdata = data;
+    struct sol_flow_node_type_http_server_string_options *opts =
+        (struct sol_flow_node_type_http_server_string_options *)options;
+
+    mdata->value.s = strdup(opts->value);
+    SOL_NULL_CHECK(mdata->value.s, -ENOMEM);
+
+    r = start_server(mdata, opts->path, string_response_cb);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    mdata->node = node;
+
+    return 0;
+
+err:
+    free(mdata->path);
+    return -1;
+}
+
+/* ----------------------------------- int ------------------------------------------- */
+
+static int
+int_response_cb(void *data, struct sol_http_request *request)
+{
+    int r = 0;
+    uint16_t idx;
+    bool send_json = false, modified = false;
+    enum sol_http_method method;
+    struct http_data *mdata = data;
+    char str[512] = { 0 };
```

Comment:
Zeroing out large array without need. Also, 512 bytes for an integer is quite an overkill...